### PR TITLE
.travis.yml: Add Linux

### LINKED ISSUE #47: Add Linux Travis CI
This would at least run `brew portable-package ruby` on Linux and ideally exercise the Dockerfile here too. CC @sjackman @alyssais who may be interested in this.

### PR DESCRIPTION
**Pull request by sjackman**
Fixes #47

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
Failing (as seems to be expected) but not badly and I'm not sure if this is even a legitimate failure.

**Comment by sjackman** [comment]
```
Error: homebrew/portable/portable-ruby: failed
</Homebrew libraries/> expected to not match
<"Homebrew libraries:\n  /home/linuxbrew/.linuxbrew/lib/libc.so.6 (glibc)\n  /home/linuxbrew/.linuxbrew/lib/libcrypt.so.1 (glibc)\n  /home/linuxbrew/.linuxbrew/lib/libdl.so.2 (glibc)\n  /home/linuxbrew/.linuxbrew/lib/libm.so.6 (glibc)\n  /home/linuxbrew/.linuxbrew/lib/libpthread.so.0 (glibc)\n  /home/linuxbrew/.linuxbrew/lib/libutil.so.1 (glibc)\n  /home/linuxbrew/.linuxbrew/lib/libgmp.so.10 (gmp)\nPossible undeclared dependencies:\n  gmp\n">.
Error: Failure while executing: brew test portable-ruby
```
https://travis-ci.org/Homebrew/homebrew-portable/jobs/278421362#L834

I've added `HOMEBREW_BUILD_FROM_SOURCE=1` to address this failure, as per https://github.com/Homebrew/homebrew-portable/blob/master/docker/Dockerfile.x86_64#L25

**Comment by MikeMcQuaid** [comment]
Thanks again @sjackman!

File: .travis.yml [hunk 1]
```diff
@@ -1,17 +1,36 @@
...
-  - sudo chown -R "$USER" "$HOMEBREW_REPOSITORY"
-  - git -C "$HOMEBREW_REPOSITORY" fetch;
-  - git -C "$HOMEBREW_REPOSITORY" reset --hard origin/master;
+  - if [ "${TRAVIS_OS_NAME}" = "osx" ]; then
+      HOMEBREW_REPOSITORY="$(brew --repo)";
+      sudo chown -R "$USER" "$HOMEBREW_REPOSITORY";
+      git -C "$HOMEBREW_REPOSITORY" fetch;
+      git -C "$HOMEBREW_REPOSITORY" reset --hard origin/master;
+    else
+      HOME=/home/linuxbrew;
```
Comment (MikeMcQuaid) [inline .travis.yml]: @sjackman Is there a reason there's `/home/linuxbrew` and `/home/.linuxbrew`?

Comment (sjackman) [reply .travis.yml]: `HOME=/home/linuxbrew` and `HOMEBREW_PREFIX=/home/.linuxbrew/.linuxbrew`

Comment (MikeMcQuaid) [reply .travis.yml]: @sjackman can you check you've not typoed that?

Comment (sjackman) [reply .travis.yml]: Yep, that's a typo. Should be
`HOME=/home/linuxbrew` and
`HOMEBREW_PREFIX=/home/linuxbrew/.linuxbrew`

Comment (MikeMcQuaid) [reply .travis.yml]: Ok, thanks.

**Pull request merged**